packages/aws-cdk-lib/aws-apigateway/test/method.test.ts

```diff
@@ -767,20 +769,137 @@ describe('method', () => {
 
     // WHEN
     new apigw.Method(stack, 'MethodAuthScopeUsed', {
-      httpMethod: 'POST',
+      httpMethod: 'OPTIONS',
       resource: api.root,
       options: {
         apiKeyRequired: true,
+        authorizationType: apigw.AuthorizationType.NONE,
         authorizationScopes: ['MethodAuthScope'],
       },
     });
 
     // THEN
-    Template.fromStack(stack).hasResourceProperties('AWS::ApiGateway::Method', {
-      ApiKeyRequired: true,
-      AuthorizationScopes: ['MethodAuthScope'],
+    expect(() => Template.fromStack(stack).hasResourceProperties('AWS::ApiGateway::Method', {
+      AuthorizationScopes: ['DefaultAuth'],
+    })).toThrow(/Template has 1 resources with type AWS::ApiGateway::Method, but none match as expected./);
+
+  });
+
+  test('Method options Auth Scopes should be None when Auth Type from Method is IAM', () => {
```

Comment:
Can some of the tests you added for the different Auth Types be combined into 1 test?
```suggestion
  test.each([
  apigw.AuthorizationType.IAM,
  apigw.AuthorizationType.NONE,
  apigw.AuthorizationType.CUSTOM,
])('...', (authType) => {
    ...
    const api = new apigw.RestApi(stack, 'test-api', {
      cloudWatchRole: false,
      deploy: false,
      defaultMethodOptions: {
        authorizationType: authType,
        authorizationScopes: ['DefaultAuth'],
      },
    });
    ....
}
```

Reply:
Good call, updated